### DISCUSSION TITLE
Example for how to load GGUF inside a Python program?

### DISCUSSION THREAD

**Discussion by atipasvanund** [ml-explore/mlx#775] [Q&A]
Is there something similar to this?

```
from mlx_lm import load

model, tokenizer = load("mode.gguf")
```

The examples I can find shows how to do it from console but not how to do it from within a Python program.

**Comment by awni** [comment]
You can use [`mx.load`](https://ml-explore.github.io/mlx/build/html/python/_autosummary/mlx.core.load.html#mlx.core.load) to load any `gguf` file. It will give you two dictionaries, one for the metadata and one for the arrays themselves.

This [function](https://github.com/ml-explore/mlx-examples/blob/main/llms/gguf_llm/models.py#L264) in MLX Examples shows how to use the GGUF to make an LLM + tokenizer.
